src/segment-loader.js

```diff
@@ -154,6 +154,159 @@ export const timestampOffsetForSegment = ({
+ * segment until that value is provided.
+ *
+ * CASE 2: Video prepares to load a new timeline before audio:
+ *
+ * Timeline:       0                 1
+ * Audio Segments: 0 1 2 3 4 5 DISCO 6 7 8 9
+ * Audio Loader:             ^
+ * Video Segments: 0 1 2 3 4 5 DISCO 6 7 8 9
+ * Video Loader                      ^
+ *
+ * In the above example, the video loader is preparing to load the 6th segment, the first
+ * after a discontinuity, while the audio loader is still loading the 5th segment, before
+ * the discontinuity.
+ *
+ * If the video loader goes ahead and loads and appends the 6th segment, then once the
+ * segment is loaded and processed, both the video and audio timestamp offsets will be
+ * set, since video is used as the primary timing source. This is to ensure content lines
+ * up appropriately, as any modifications to the video timing are reflected by audio when
+ * the video loader sets the audio and video timestamp offsets to the same value. However,
+ * setting the timestamp offset for audio before audio has had a chance to change
+ * timelines will likely lead to desyncing, as the audio loader will append segment 5 with
+ * a timestamp intended to apply to segments from timeline 1 rather than timeline 0.
+ *
+ * CASE 3: When seeking, audio prepares to load a new timeline before video
+ *
+ * Timeline:       0                 1
+ * Audio Segments: 0 1 2 3 4 5 DISCO 6 7 8 9
+ * Audio Loader:           ^
+ * Video Segments: 0 1 2 3 4 5 DISCO 6 7 8 9
+ * Video Loader            ^
+ *
+ * In the above example, both audio and video loaders are loading segments from timeline
+ * 0, but imagine that the seek originated from timeline 1.
+ *
+ * When seeking to a new timeline, the timestamp offset will be set based on the expected
+ * segment start of the loaded video segment. In order to maintain sync, the audio loader
+ * must wait for the video loader to load its segment and update both the audio and video
+ * timestamp offsets before it may load and append its own segment. This is the case
+ * whether the seek results in a mismatched segment request (e.g., the audio loader
+ * chooses to load segment 3 and the video loader chooses to load segment 4) or the
+ * loaders choose to load the same segment index from each playlist, as the segments may
+ * not be aligned perfectly, even for matching segment indexes.
+ *
+ * @param {Object} timelinechangeController
+ * @param {number} currentTimeline
+ *        The timeline currently being followed by the loader
+ * @param {number} segmentTimeline
+ *        The timeline of the segment being loaded
+ * @param {('main'|'audio')} loaderType
+ *        The loader type
+ * @param {boolean} audioDisabled
+ *        Whether the audio is disabled for the loader
+ *
+ * @return {boolean}
+ *         Whether the loader should wait for a timeline change from the timeline change
+ *         controller before processing the segment
+ */
+export const shouldWaitForTimelineChange = ({
```

Comment:
does this account for something like the audio missing a segment? It would previously cause lots of av sync issues and something that squarebracket was working on. If not, we should remember to circle back to this case later on, we don't need to solve it in this PR.
I think the example would end up looking kind of like:
```

 * Timeline:       0                 1
 * Audio Segments:     2 3 4 5 DISCO 6 7 8 9
 * Audio Loader:           ^
 * Video Segments: 0 1 2 3 4 5 DISCO 6 7 8 9
 * Video Loader            ^
```

Reply:
Do we have any example content for it? And would this just be for the live case where the playlists are updating out-of-sync?

Reply:
I don't think we have one but I think squarebracket said we can take a demuxed stream and delete the first segment from the audio playlist but leave the video playlist alone to demonstrate the issue.

Reply:
Tested, and TBA alone fixed the issue.